virttest/libvirt_xml/devices/address.py

```diff
@@ -46,3 +54,33 @@ def new_from_element(cls, element, virsh_instance=base.base.virsh):
                 "type attribute is manditory for " "Address class"
             )
         return cls.new_from_dict(edict, virsh_instance=virsh_instance)
+
+    class Zpci(base.base.LibvirtXMLBase):
+        """Represents the optional subelement for zpci addresses"""
+
+        __slots__ = ("uid", "fid")
+
+        def __init__(self, virsh_instance=base.base.virsh):
+            accessors.XMLAttribute(
+                "uid",
+                self,
+                parent_xpath="/",
+                tag_name="zpci",
+                attribute="uid",
+            )
+            accessors.XMLAttribute(
+                "fid",
+                self,
+                parent_xpath="/",
+                tag_name="zpci",
+                attribute="fid",
+            )
+            super(self.__class__, self).__init__(virsh_instance=virsh_instance)
+            self.xml = "<zpci/>"
+
+        @classmethod
+        def new_from_dict(cls, attributes, virsh_instance=base.base.virsh):
```

Comment:
I'm thinking we could have a base function of this kind : new_from_dict() for all xml classes, which could be implemented really simple by:
```
inst = class()
inst.setup_attrs(**attrs)
return inst
```
This way, we could use this one-line function to create xml instance without having to create a new new_from_dict function for each class

Reply:
maybe new_from_attrs() to avoid conflict

Reply:
@chloerh That base class, shouldn't that be dealt with in another PR? IIUC, it will require other classes to be modified, too. Do I understand this correctly?

Reply:
Yes. I forgot to update my comment. Actually I found that there is a function new_from_dict for all device classes. Anyway, this pr is ok for me.